src/test/regress/sql/pg17.sql

```diff
@@ -180,3 +177,46 @@ DROP SCHEMA pg17_corr_subq_folding CASCADE;
 
 -- PG17-specific tests go here.
 --
+
+CREATE SCHEMA pg17;
+SET search_path to pg17;
+
+-- MAINTAIN privilege tests
+
+CREATE ROLE regress_maintain;
+CREATE ROLE regress_no_maintain;
+
+ALTER ROLE regress_maintain WITH login;
+GRANT USAGE ON SCHEMA pg17 TO regress_maintain;
+ALTER ROLE regress_no_maintain WITH login;
+GRANT USAGE ON SCHEMA pg17 TO regress_no_maintain;
+
+SET citus.shard_count TO 1; -- For consistent remote command logging
+CREATE TABLE dist_test(a int, b int);
+SELECT create_distributed_table('dist_test', 'a');
+INSERT INTO dist_test SELECT i % 10, i FROM generate_series(1, 100) t(i);
+
+GRANT MAINTAIN ON dist_test TO regress_maintain;
+
+SET ROLE regress_no_maintain;
+
+SET citus.log_remote_commands TO ON;
+ANALYZE dist_test;
+VACUUM dist_test;
+RESET citus.log_remote_commands;
+
+SET ROLE regress_maintain;
+
+SET citus.log_remote_commands TO ON;
+ANALYZE dist_test;
+VACUUM dist_test;
+RESET citus.log_remote_commands;
+
+RESET ROLE;
+
+DROP SCHEMA pg17 CASCADE;
+DROP ROLE regress_maintain;
+DROP ROLE regress_no_maintain;
+RESET citus.next_shard_id;
+RESET citus.shard_count;
+RESET citus.shard_replication_factor;
```

Comment:
endline, and reordering suggestion

```suggestion
RESET ROLE;

RESET citus.log_remote_commands;
RESET citus.next_shard_id;
RESET citus.shard_count;
RESET citus.shard_replication_factor;

DROP SCHEMA pg17 CASCADE;
DROP ROLE regress_maintain;
DROP ROLE regress_no_maintain;

```